### PR TITLE
Create workflow for monorepo CI build

### PR DESCRIPTION
**Pull request by kerbou**

### CODE REVIEW

File: .github/workflows/ci-orchestrator.yml [hunk 1]
```diff
@@ -32,18 +34,54 @@ jobs:
...
+    uses: ./.github/workflows/ci-dotnet.yml
+    with:
+      release_name: process_manager_dbmigrations
+      project_file: source/ProcessManager.DatabaseMigration/ProcessManager.DatabaseMigration.csproj
+
+  ci_run_tests:
+    needs: changes
+    uses: ./.github/workflows/ci-dotnet-runtests.yml
+    with:
+      print_debug_output: 'true'  # TODO: Remove before merge
```
Comment (djorgensendk) [inline .github/workflows/ci-orchestrator.yml:82]: remember this

File: .github/workflows/ci-dotnet-runtests.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Find testprojects in affected projects
+        id: set-matrix
+        shell: pwsh
+        run: |
+          $affectedProjects = '${{ inputs.affected_changes  }}' | ConvertFrom-Json
+          # $matrixBuilds = $affectedProjects | Where-Object {$_.Name -like '*Tests*'} | Select-Object -ExpandProperty Filepath  | ConvertTo-Json -Compress
```
Comment (djorgensendk) [inline .github/workflows/ci-dotnet-runtests.yml:50]: commented

File: .github/workflows/ci-dotnet-runtests.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+          # I couldn't find a better way to lowercase keys in a hashtable
+          $matrixBuilds = @()
+
+          foreach ($project in $affectedProjects) {
+            if ($project.Name -like '*Tests*') {
+              $matrixBuilds += [PSCustomObject]@{
+                name           = $project.Name
+                filepath       = $project.FilePath
+                assembly_name  = "Energinet.DataHub.$($project.Name).dll"
+                dotnet_version = "net8.0"
```
Comment (djorgensendk) [inline .github/workflows/ci-dotnet-runtests.yml:61]: should this be a parameter instead?

File: .github/workflows/ci-dotnet-runtests.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  #           --output '${{ env.OUTPUT_PATH }}' \
+  #           -p:ErrorOnDuplicatePublishOutputFiles=false  # To avoid the error "Found multiple publish output files with the same relative path"
+
+  #         # ls ${{ env.OUTPUT_PATH}}
+
+  #     - uses: actions/upload-artifact@v4
+  #       with:
+  #         retention-days: 2
+  #         name: pm-tests-${{ matrix.project.name }}-${{ github.event.pull_request.number }}
+  #         path: ${{ env.OUTPUT_PATH }}/*
```
Comment (djorgensendk) [inline .github/workflows/ci-dotnet-runtests.yml:115]: commented

File: .github/workflows/ci-dotnet-runtests.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+  integration_tests:
+    needs: [
+      build_matrix
+    ]
+    strategy:
+      fail-fast: false
+      matrix:
+        project: ${{fromJson(needs.build_matrix.outputs.matrix_builds)}}      
+    uses: Energinet-DataHub/.github/.github/workflows/dotnet-postbuild-test-monorepo.yml@kerbou/2524-build-brs-pipeline
```
Comment (djorgensendk) [inline .github/workflows/ci-dotnet-runtests.yml:125]: use correct branch

File: .github/workflows/processmanager-client-bundle-publish.yml [hunk 6]
```diff
@@ -156,22 +156,3 @@ jobs:
...
-      always()
-    steps:
-      - name: Verify if merge is allowed
-        run: |
-          echo "${{ toJSON(needs) }}"
-
-          if [[ ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }} = true ]]; then
-              echo "Failed"
-              exit 1
-          fi
```
[the remark this answers is not in the document]
Comment (kerbou) [reply .github/workflows/processmanager-client-bundle-publish.yml:177]: It is now covered by allow_merge in ci-orchestrator.yml

File: .github/workflows/ci-dotnet-runtests.yml (hunk 3 above)
Comment (kerbou) [reply .github/workflows/ci-dotnet-runtests.yml:61]: Fixed

File: .github/workflows/ci-dotnet-runtests.yml (hunk 4 above)
Comment (kerbou) [reply .github/workflows/ci-dotnet-runtests.yml:115]: Will leave it for now to test relative paths in TestCommon in later PR

**Review by lanni-energinet — APPROVED** [review]

File: .github/workflows/ci-dotnet-runtests.yml [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  #           --output '${{ env.OUTPUT_PATH }}' \
+  #           -p:ErrorOnDuplicatePublishOutputFiles=false  # To avoid the error "Found multiple publish output files with the same relative path"
+
+  #         # ls ${{ env.OUTPUT_PATH}}
+
+  #     - uses: actions/upload-artifact@v4
+  #       with:
+  #         retention-days: 2
+  #         name: pm-tests-${{ matrix.project.name }}-${{ github.event.pull_request.number }}
+  #         path: ${{ env.OUTPUT_PATH }}/*
```
Comment (dstenroejl) [inline .github/workflows/ci-dotnet-runtests.yml:119]: Can this be deleted, or why is it commented?

Comment (kerbou) [reply .github/workflows/ci-dotnet-runtests.yml:119]: It is intended

**Pull request merged**